controllers/vdoconfig_controller.go

```diff
@@ -663,32 +620,70 @@ func (r *VDOConfigReconciler) reconcileNodeLabel(ctx vdocontext.VDOContext, req
 	return nil
 }
 
-func (r *VDOConfigReconciler) reconcileNodeProviderID(ctx vdocontext.VDOContext, config *vdov1alpha1.VDOConfig, clientset *kubernetes.Clientset) (*vdov1alpha1.VDOConfig, error) {
+func (r *VDOConfigReconciler) reconcileNodeProviderID(ctx vdocontext.VDOContext, config *vdov1alpha1.VDOConfig, clientset *kubernetes.Clientset, vsphereCloudConfigs *[]vdov1alpha1.VsphereCloudConfig) (*vdov1alpha1.VDOConfig, error, []v1.Node) {
 	nodeStatus := make(map[string]vdov1alpha1.NodeStatus)
-	cpiStatus := vdov1alpha1.Configured
+	var NodeList []v1.Node
 
+	cpiStatus := vdov1alpha1.Configured
 	nodes, err := clientset.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
 	if err != nil {
-		return config, errors.Wrapf(err, "Unable to fetch list of nodes")
+		return config, errors.Wrapf(err, "Error reconciling the providerID for CPI, unable to fetch list of nodes"), NodeList
 	}
 
+nodeLoop:
 	for _, node := range nodes.Items {
 		if len(node.Spec.ProviderID) > 0 {
 			nodeStatus[node.Name] = vdov1alpha1.NodeStatusReady
-		} else {
-			nodeStatus[node.Name] = vdov1alpha1.NodeStatusPending
-			cpiStatus = vdov1alpha1.Configuring
+			NodeList = append(NodeList, node)
+			r.Logger.Info(fmt.Sprintf("Added %s node to NodeList", node.Name))
+			continue nodeLoop
 		}
+
+		for _, taint := range node.Spec.Taints {
+			if taint.Key == CLOUD_PROVIDER_INIT_TAINT_KEY {
+
+				if val, ok := NodeAvailabilityMap[node.Name]; ok {
+					if val {
+						nodeStatus[node.Name] = vdov1alpha1.NodeStatusPending
+						cpiStatus = vdov1alpha1.Configuring
+						NodeList = append(NodeList, node)
```

Comment:
what is the difference between NodeAvailbilityMap and NodeList array

Reply:
Earlier NodeAvailabilityMap was used  to check if the node exists in vspherecloudconfig and NodeList for having all the nodes meant to be CPI configured. 
Currently, merged both into NodeAvailabilityMap.